### PR TITLE
[RFR] Fixed bootstrap upgrade and webpack devtool

### PR DESCRIPTION
**Pull request by djhi**
- [x] Fixed incorrect bootstrap classes following upgrade to alpha 5
- [x] Upgraded webpack to new SourceMapDevToolPlugin to ensure correct sourcemap on dev
- [x] Added a yarn.lock file

### CODE REVIEW

File: src/frontend/js/app/AppNavbar/UserNavbarItem.js [hunk 1]
```diff
@@ -1,23 +1,34 @@
...
-            </a>
-            <div className="dropdown-menu">
-                <a className="dropdown-item" href="/sign-out" onClick={signOut} role="button">Sign out</a>
-            </div>
-        </li>
-    </ul>
-);
+class UserNavbarItem extends Component {
+    signOut = (event) => {
+        event.preventDefault();
```
Comment (ThieryMichel) [inline src/frontend/js/app/AppNavbar/UserNavbarItem.js]: What does the preventDefault prevent? The href? If so why put a link with an href in the first place?

Comment (djhi) [inline src/frontend/js/app/AppNavbar/UserNavbarItem.js]: to make eslint happy :)

Comment (djhi) [inline src/frontend/js/app/AppNavbar/UserNavbarItem.js]: oh wait, I just noticed it's now possible in alpha 5. Let me try

Comment (djhi) [inline src/frontend/js/app/AppNavbar/UserNavbarItem.js]: confirmed and applied

File: package.json [hunk 2]
```diff
@@ -116,6 +116,7 @@
     "webpack-dev-server": "2.1.0-beta.9"
   },
   "scripts": {
+    "postinstall": "make install-selenium",
```
Comment (Kmaschta) [inline package.json]: :+1: 
Can you remove this instruction in the makefile then?

File: src/admin/js/Layout.js [hunk 3]
```diff
@@ -8,7 +8,9 @@ import Notification from 'admin-on-rest/lib/mui/layout/Notification';
 import LayoutTitle from './LayoutTitle';
 
 const Layout = ({ isLoading, children }) => {
-    const RightElement = isLoading ? <CircularProgress color="#fff" size={0.5} /> : <span />;
+    const RightElement = isLoading
+        ? <CircularProgress color="#fff" size={30} thickness={2} style={{ margin: 8 }} />
+        : <span />;
```
Comment (Kmaschta) [inline src/admin/js/Layout.js]: Is there a way to not show an empty span most of the time?

File: src/api/lib/middlewares/pgCrud.js [hunk 4]
```diff
@@ -36,7 +36,7 @@ export default (queriesFactory, configuredMethods = {}) => {
             });
 
             ctx.body = await queries.selectPage(query.limit, query.offset, query.filter, query._sort, query._sortDir, other); // eslint-disable-line no-underscore-dangle
-            const totalCount = (ctx.body[0]) ? ctx.body[0].totalcount : (await queries.countAll());
+            const totalCount = (ctx.body[0]) ? ctx.body[0].totalcount : (await queries.countAll()).count;
```
Comment (Kmaschta) [inline src/api/lib/middlewares/pgCrud.js]: Would be great to have a test for that

File: package.json (hunk 2 above)
Comment (Kmaschta) [inline package.json]: But ... This will install selenium in the production server, so I'm not in favor of this.

Comment (djhi) [inline package.json]: I'd like to be sure selenium is correctly installed each time I do an `npm install`. Would it be ok to add a condition checking for `development` env in the makefile ?

File: src/admin/js/Layout.js (hunk 3 above)
Comment (djhi) [inline src/admin/js/Layout.js]: No, that's the way it is done in admin-on-rest currently. I haven't tested it but I suspect material-ui to have a funny behavior without it

**Comment by djhi** [comment]
I took the opportunity to also propose a way to make our API more testable. Have a look at the pgCrudFactory [here](https://github.com/marmelab/javascript-boilerplate/pull/139/commits/aec1d17457f292454aefeb9f638cd45fdcc08cfb#diff-43c168bc2fd6920c796ed6645619f2a3R12)

File: src/admin/js/Layout.js (hunk 3 above)
Comment (djhi) [inline src/admin/js/Layout.js]: Well, after tests, it seems it is not needed indeed

**Comment by alexisjanvier** [comment]
👍  for Yarn :)
But I'll have to ask @ThieryMichel a quick presentation of the `coPostgresQuery` lib in order to give an opinion the `pgCrudFactory ` refactoring ... What is sure, is that I really don't like this `ctx`name! What does it mean ?

**Comment by djhi** [comment]
@alexisjanvier `ctx` is from `koa`. It's the context.

**Comment by alexisjanvier** [comment]
@djhi ok :) Thanks

**Comment by djhi** [comment]
The refactoring has nothing to do with `coPostgresQuery`. I just extracted the methods previously bound directly to the koa routes in another object to make testing easier

**Pull request merged**